### ISSUE TITLE
par and limit extractor functions

### ISSUE THREAD

**Issue by ldecicco-USGS** [USGS-R/gsplot#246]
```
gs <- gsplot() %>%
  points( y=c(3,1,2), x=1:3, xlim=c(0,NA),ylim=c(0,NA),
          col="blue", pch=18, legend.name="Points", xlab="Index") %>%
  lines(c(3,4,3), c(2,4,6), legend.name="Lines", ylab="Data") %>%
  abline(b=1, a=0, legend.name="1:1") %>%
  legend(location="topleft",title="Awesome!") %>%
  grid() %>%
  bgCol(col="lightgrey") %>%
  error_bar(x=1:3, y=c(3,1,2), y.high=c(0.5,0.25,1), y.low=0.1) %>%
  arrows(x0=0.75, y0=2, x1=1, y1=2.8, lwd=2) %>%
  title("Graphing Fun") %>%
  text(.75,1.75,labels="Weird data") 
gs
u <- par("plt")
v <- c( (u[1]+u[2])/2, u[2], (u[3]+u[4])/2, u[4] )
config("par",new=TRUE,fig=v,mar=c(1,1,1,1))
gs2 <- gsplot() %>%
  points(1, 2) %>%
  bgCol(col="white")
gs2
# To get back to the 'default par':
par(gs.par(gs))
```

Suggestions?

**Comment by ldecicco-USGS** [comment]
ie. `par.gs`

**Comment by ldecicco-USGS** [comment]
Others?
xlim.gs
ylim.gs
par.gs

**Comment by jordansread** [comment]
`log.gs`

**Comment by jordansread** [comment]
Anyone started on this? If not, I might take a shot. Flight is delayed :airplane:

**Comment by ldecicco-USGS** [comment]
Of course it is. Do it. These will be uber-handy functions.

**Comment by jordansread** [comment]
:airplane: ORD :airplane: ery-time

**Comment by lindsayplatt** [comment]
these will be super handy for some repgen features too

**Comment by jordansread** [comment]
``` R
 usrDef <- gsplot(mar=c(4,4,4,4), xaxs='r', yaxs='r') %>% 
    points(x=1, y=2, side=c(3,2), legend.name="Points 1", cex=3, xlab='cat')

xlim(usrDef)[[1]]
c(1,1)

  usrDef <- points(usrDef,x=3:10,y=4:11, side=c(3,2))

xlim(usrDef)[[1]]
c(1,10)
```

**Comment by jordansread** [comment]
``` r
usrDef <- gsplot(mar=c(4,4,4,4), xaxs='r', yaxs='r') %>% 
     points(x=1, y=2, side=c(3,2), legend.name="Points 1", cex=3, xlab='cat')
par(usrDef)
$mar
[1] 4 4 4 4

$xaxs
[1] "r"

$yaxs
[1] "r"

$tcl
[1] 0.3

$mgp
[1] 1.5 0.3 0.0
```

**Comment by lindsayplatt** [comment]
Cool! Would we want the function to be called `par.gs` as discussed above?

**Comment by jordansread** [comment]
Oh, I went with `par` instead (and `xlim` etc) instead of `xlim.gs` and `par.gs`. Should it be `par.gs`? Wasn't sure that this was necessary

**Comment by jordansread** [comment]
see #247 ? @lindsaycarr and @ldecicco-USGS feedback?

**Comment by ldecicco-USGS** [comment]
I'd like the workflow to be exactly what Jordan has above. The .gs I thought was an implementation detail (a gs method for par and xlim)...not something the user should have to type. And, it was just a quick thought when writing the comment, I hadn't actually worked out all the logic to know if those were methods or not that could have a gs component. 

The only part that gives me pause is:

```
xlim(usrDef)[[1]]
c(1,10)
```

Can it also be named? Like:

```
xlim(usrDef)$side1
c(1,10)
```

or something....the index of the view seems irrelevant to the user, but the side is a concept they should understand.

**Comment by jordansread** [comment]
Great idea LDC. I will add that. 

What about something for `log`?

**Comment by ldecicco-USGS** [comment]
Makes sense to me.

**Comment by jordansread** [comment]
so `log` will still come back as `view` `view` etc, or is there a better naming convention for the list? maybe `side.1.2`?

**Comment by jordansread** [comment]
maybe that should be how views are named...

**Comment by ldecicco-USGS** [comment]
Hmm....I didn't actually run your code. So right now, log is "x", "y", "xy", or ""? Could we do:

```
log(usrDef)$side.1
TRUE
```

or do you want to preserve the string?

**Comment by jordansread** [comment]
I was thinking that it would be the same type of input you would put into the argument. So xlim=c(1,2) and log="xy". Could have it be more helpful like this:

``` r
log(usrDef, side=1)
[1] TRUE
log(usrDef)
[1] "xy"
```

but that may be confusing to get logical on some and string on other outputs. Thoughts?

**Comment by ldecicco-USGS** [comment]
I like it. It's a tough one because of the way base R takes the input. Is it even possible to make a graph where side 2 is log and side 4 is linear? If so, we'd need to specify the view:

``` r
log(usrDef, side=1)
[1] TRUE
log(usrDef)$sides.1.2
[1] "xy"
```

I'm digging the idea of naming the views based on their sides...for lots o'things.

**Comment by jordansread** [comment]
OK, but just to clarify, here is the expected behavior for me:

``` r
log(usrDef, side=c(1,2))
[1] TRUE TRUE
log(usrDef)$sides.1.2
[1] "xy"
```

maybe confusing?

**Comment by jordansread** [comment]
switching this up a bit to be `logged` instead of `log` because of the clash in the generic for `log`, which has formals `x` and `base` because it is a math operation and doesn't make sense here. 

so the above is

``` r
logged(usrDef, side=c(1,2))
[1] TRUE TRUE
logged(usrDef)$sides.1.2
[1] "xy"
```

**Comment by jordansread** [comment]
closed via #257
